File: dr-multi-drm-sdk/dr_multi_drm_sdk_android/android/src/main/kotlin/com/doverunner/dr_multi_drm_sdk_android/DownloadContentHandler.kt
package com.doverunner.dr_multi_drm_sdk_android

import com.doverunner.dr_multi_drm_sdk_android.sdk.WvSdk
import android.content.Context
import com.doverunner.dr_multi_drm_sdk_android.event.DownloadProgressEventImpl
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class DownloadContentHandler: EventChannel.StreamHandler {
    private var channel: EventChannel? = null
    private var context: Context? = null

    fun startListening(context: Context, messenger: BinaryMessenger) {
        if (channel != null) {
            stopListening()
        }

        channel = EventChannel(messenger, "com.doverunner.drmsdk/download_progress")
        channel?.setStreamHandler(this)
        this.context = context

//        try {
//            DownloadService.start(context, DownloadContentService::class.java)
//        } catch (e: IllegalStateException) {
//            DownloadService.startForeground(context, DownloadContentService::class.java)
//        }
    }

    fun stopListening() {
        channel?.setStreamHandler(null)
        channel = null
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        context?.let {
//            PallyConSdk.getInstance(it).setDownloadProgress(events)
            WvSdk.getInstance(it).setDownloadProgressEvent(DownloadProgressEventImpl(events))
        }

    }

    override fun onCancel(arguments: Any?) {
        context?.let {
            WvSdk.getInstance(it).setDownloadProgressEvent(null)
        }
    }
}

File: dr-multi-drm-sdk/dr_multi_drm_sdk_ios/ios/Classes/handlers/DownloadContentHandler.swift
import Flutter
import Foundation

public class DownloadContentHandler: NSObject, FlutterStreamHandler {
    private var events: FlutterEventSink?

    public func startListening() {
        
    }

    public func stopListening() {
//        events?.setStreamHandler(null)
//        events = null
    }

    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        // event 연결
        self.events = events
        DrMultiSdk.shared.setDownloadProgress(eventSink: events)
        return nil
    }

    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        // event 연결 해제
        DrMultiSdk.shared.setDownloadProgress(eventSink: nil)
        events = nil
        return nil
    }
}

